tests/test_functions.py

```diff
@@ -30,6 +31,14 @@ def function_d(d, d1, d2=1):
     return d + d1 + d2
 
 
+def lambda_a():
+    pkl = os.path.join(
+        os.path.dirname(__file__),
+        "lambda.pkl")
```

Comment:
It would be much better to write the file contents elsewhere, instead of relying on a stored pickle file.  Was the file written in python 3.7?  `dill` tests are currently run with 2.7, 3.6, 3.7, 3.8, 3.9, 3.10, pypy27, pypy36, and pypy37.  Is the test only supposed to run with 3.8?

You don't need to add a test case into the code at the moment, if it's difficult.  Rather, just present the details in the main conversation of the Github issue, so I and others can reproduce what you are seeing.

Reply:
The file was written with python 3.7 /  dill 0.3.0. I think that’s why you can’t put the file contents there (and need the binary) - the bug seems to be that dill isn’t backwards compatible. 

The pickled object is just lambda x: x.

Reply:
Here's an example using lambda in `dill` master with 3.7:
```
Python 3.7.10 (default, Mar 18 2021, 06:11:04) 
[Clang 9.0.0 (clang-900.0.39.2)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import dill
>>> dill.dump(lambda x:x, open('test.pkl', 'wb'))
```
and loading with 3.8...
```
Python 3.8.10 (default, May  7 2021, 23:18:56) 
[Clang 9.0.0 (clang-900.0.39.2)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import dill
>>> f = dill.load(open('test.pkl', 'rb'))
>>> f(4)
4
>>> 
```
What is your PR doing that is not possible currently?

Reply:
You need to dump with dill 0.3.0 and python 3.7, not master/3.7 to reproduce.

Reply:
So... you are saying that the issue is that old pickles from python 3.7 created with `dill` 0.3.0 don't unpickle in python 3.8 with `dill` master.  I'm assuming this is also the case for other old versions of `dill` (before `_create_function`  was recently modified).

Am I correct in thinking that you could, as a workaround, load the pickle in 3.7 with `dill` master, and then dump it again... then the resulting file would be able to be opened with 3.8?

Reply:
My 3.8 workaround is just to set:

 dill._dill._reverse_typemap[‘CodeType’] = dill._dill._create_code

much easier than re-serialising all the pickle files I have lying around :)

Reply:
Riiight, of course... hence the PR.  I never really considered it, however, I'm wonder if adding other of the `_create_` functions to the `reverse_typemap` is worth investigating.  I'm not certain of what functionality it might impact.